common/log/slog.go

```diff
@@ -0,0 +1,204 @@
+// The MIT License
+//
+// Copyright (c) 2023 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Part of this implementation was taken from https://github.com/uber-go/zap/blob/99f1811d5d2a52264a9c82505a74c2709b077a73/exp/zapslog/handler.go
+
+package log
+
+import (
+	"context"
+	"log/slog"
+
+	"go.temporal.io/server/common/log/tag"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type handler struct {
+	zapLogger *zap.Logger
+	logger    Logger
+	attrs     []slog.Attr
+	group     string
+}
+
+var _ slog.Handler = (*handler)(nil)
+
+// NewSlogLogger creates an slog.Logger from a given logger.
+func NewSlogLogger(logger Logger) *slog.Logger {
+	logger = withIncreasedSkip(logger, 3)
+	return slog.New(&handler{logger: logger, zapLogger: extractZapLogger(logger), group: "", attrs: nil})
+}
+
+// Enabled reports whether the handler handles records at the given level.
+func (h *handler) Enabled(_ context.Context, level slog.Level) bool {
+	if h.zapLogger == nil {
+		return true
+	}
+	return h.zapLogger.Core().Enabled(convertSlogToZapLevel(level))
+}
+
+// Handle implements slog.Handler.
+func (h *handler) Handle(_ context.Context, record slog.Record) error {
```

Comment:
Not capturing the log location and stack trace here. We seem to not need this functionality since our zapLogger adds the `logging-call-at` tag.